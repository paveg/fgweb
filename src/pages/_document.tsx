import Document, { Html, Head, Main, NextScript } from 'next/document';

class CustomDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html lang="ja">
        <Head>
          <meta name="Description" content="This is a Funai Gaming website that mainly describes teams and activities." />
          <link rel="dns-prefetch" href="https://www.google.co.jp" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default CustomDocument
