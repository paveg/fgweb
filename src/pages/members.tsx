import React from 'react'
import { Layout } from '../components/layout'
import TwitterIcon from '@material-ui/icons/Twitter'
import { IconButton } from '../components/parts/buttons/iconButton'

type Member = {
  nickname: string
  twitterURL: string
}

const members: Member[] = [
  {
    nickname: 'funai',
    twitterURL: 'https://twitter.com/paveg_',
  },
]

export const Parts: React.FC = (): JSX.Element => (
  <Layout pageTitle="Members">
    <div className="w-full text-center mx-auto">
      <table className="rounded-t-lg m-5 w-5/6 mx-auto bg-gray-900 text-gray-200">
        <thead>
          <tr className="text-center border-b border-gray-300">
            <th className="px-4 py-3">Name</th>
            <th className="px-4 py-3">SNS</th>
          </tr>
        </thead>
        <tbody>
          {members.map((member: Member) => {
            return (
              <tr
                key={member.nickname}
                className="bg-white text-gray-900 border-b border-gray-700"
              >
                <td className="px-4 py-3">{member.nickname}</td>
                <td className="px-4 py-3">
                  <IconButton
                    small
                    label="Go to Twitter profile page."
                    apperance="tertiary"
                    IconComponent={TwitterIcon}
                    href="https://twitter.com/paveg_"
                    target="_blank"
                  />
                </td>
              </tr>
            )
          })}
        </tbody>
      </table>
    </div>
  </Layout>
)

export default Parts
