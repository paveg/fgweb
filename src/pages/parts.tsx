import React from 'react'
import { Layout } from '../components/layout'
import { Button } from '../components/parts/buttons/button'

export const Parts: React.FC = (): JSX.Element => (
  <Layout pageTitle="Parts">
    <div className="w-full text-center mx-auto">
      <h2>Buttons</h2>
      <Button apperance="primary">Primary</Button>
      <Button apperance="primary" small>
        Small Primary
      </Button>
      <Button apperance="secondary">Secondary</Button>
      <Button apperance="neutral">Neutral</Button>
      <Button
        apperance="error"
        onClick={() => {
          return window.alert('alert opened.')
        }}
      >
        Alert
      </Button>
      <Button apperance="error" disabled>
        Disabled
      </Button>
    </div>
  </Layout>
)

export default Parts
