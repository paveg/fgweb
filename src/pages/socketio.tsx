import { useEffect } from 'react'
import * as React from 'react'
import io from 'socket.io-client'

const useChatService = (initialMessage) => {
  const [messages, setMessages] = React.useState([initialMessage])

  const socketRef = React.useRef()

  useEffect(() => {
    socketRef.current = io('http://localhost:3001')
    socketRef.current.on('broadcast', (payload) => {
      setMessages((prevMessages) => [...prevMessages, payload])
    })
    return () => {
      socketRef.current.disconnect()
    }
  }, [])

  const sendMessage = (name, text) => {
    const newMessage = {
      name: name,
      text: text,
    }
    socketRef.current.emit('send', newMessage)
    setMessages((prevMessages) => [...prevMessages, newMessage])
  }

  return [messages, sendMessage]
}

const SocketIO: React.FC = () => {
  const [name, setName] = React.useState('noname')
  const [messages, sendMessage] = useChatService({
    name: name,
    text: `Hello, ${name}!`,
  })
  const [text, setText] = React.useState('')

  const handleOnChange = (e) => {
    setText(e.target.value)
  }
  const handleButtonClick = () => {
    sendMessage(name, text)
    setText('')
  }

  return (
    <>
      <div className="input">
        <input
          type="text"
          placeholder="name"
          onChange={(e) => {
            setName(e.target.value)
          }}
          value={name}
        />
        <input
          type="text"
          placeholder="message"
          value={text}
          onChange={handleOnChange}
        />
        <button disabled={!text} onClick={handleButtonClick}>
          Send
        </button>
      </div>
      <ul>
        {messages.map((message, index) => {
          return (
            <li key={index}>
              {message.name}: {message.text}
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default SocketIO
