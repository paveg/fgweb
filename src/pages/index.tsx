import React from 'react';
import { Layout } from '../components/layout';
import { InlineLink } from '../components/parts/inlineLink'
import { Counter } from '../components/parts/counter'

export const Home: React.FC = (): JSX.Element => {
  return <Layout pageTitle="Home">
    <Counter />

    <InlineLink href={'/parts'}>Parts</InlineLink>
  </Layout>
}

export default Home
