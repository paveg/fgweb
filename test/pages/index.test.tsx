import React from 'react'
import { render, fireEvent } from '../testUtils'
import { Home } from '../../src/pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('Alert'))
    expect(window.alert).toHaveBeenCalledWith('alert opened.')
  })
})
