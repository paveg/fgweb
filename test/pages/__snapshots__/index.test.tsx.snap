// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="body-font bg-gray-900"
  >
    <div
      class="container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center"
    >
      <a
        class="flex title-font font-medium items-center mb-4 md:mb-0"
        href="/"
      >
        <span
          class="text-xl text-white"
        >
          Funai Gaming Web
        </span>
      </a>
      <nav
        class="md:ml-auto md:mr-auto flex flex-wrap items-center text-base justify-center"
      >
        <a
          class="mr-5 hover:text-white text-gray-300"
          href="/"
        >
          Home
        </a>
        <a
          class="mr-5 hover:text-white text-gray-300"
          href="/members"
        >
          Member
        </a>
        <a
          class="mr-5 hover:text-white text-gray-300"
          href="/parts"
        >
          Parts
        </a>
      </nav>
    </div>
  </header>
  <div
    class="container mx-auto px-5 py-8 flex items-center sm:flex-row flex-col"
  >
    <main>
      <h1>
        Home
      </h1>
    </main>
  </div>
  <footer
    class="text-gray-600 body-font"
  >
    <div
      class="container px-5 py-8 mx-auto flex items-center sm:flex-row flex-col"
    >
      <a
        class="flex title-font font-medium items-center md:justify-start justify-center text-gray-900"
        href="https://fg.vercel.app/"
      >
        <span
          class="ml-3 text-xl sm:text-center"
        >
          Funai Gaming
        </span>
      </a>
      <p
        class="text-sm text-gray-500 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-200 sm:py-2 sm:mt-0 mt-4"
      >
        © 2021 Funai —
        <a
          class="text-gray-600 ml-1"
          href="https://twitter.com/paveg_"
          rel="noopener noreferrer"
          target="_blank"
        >
          @paveg_
        </a>
      </p>
      <span
        class="inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start"
      >
        <a
          aria-label="Go to Instagram profile page."
          class="hover:bg-gray-200 text-gray-900 px-4 py-2 text-xs disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          href="https://www.instagram.com/_pavlog/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
            />
          </svg>
        </a>
        <a
          aria-label="Go to Twitter profile page."
          class="hover:bg-gray-200 text-gray-900 px-4 py-2 text-xs disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          href="https://twitter.com/paveg_"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
        </a>
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
